src/libs/ajax/AzureStorage.ts

```diff
@@ -64,21 +79,29 @@ export const AzureStorage = signal => ({
     const xml = new window.DOMParser().parseFromString(text, 'text/xml')
     const blobs = _.map(
       blob => ({
-        name: _.head(blob.getElementsByTagName('Name')).textContent,
+        name: _.head(blob.getElementsByTagName('Name'))?.textContent || '',
         lastModified: new Date(
-          _.head(blob.getElementsByTagName('Last-Modified')).textContent
+          _.head(blob.getElementsByTagName('Last-Modified'))?.textContent || '0'
```

Comment:
I prefer setting into an incorrect date since the only case in which this will be 0 is if the azure storage return payload changes, and having this always be a number ( as opposed to optional) allows for more easy processing.

Reply:
This comment becomes irrelevant with moving this logic to the `listNotebooks` function though

Reply:
I have a strong objection to "magic values" in general.  I think @nawatts suggestion to set to null in case of invalid/missing is highly preferred since it is much more explicit.  Even though this comment is outdated, I would suggest the new place that this logic lives in follow guidance similar to Nick's suggestion.  We WANT the consuming/visualizing code to be forced to deal with edge cases, not have a "forgiving" value hide things.